controllers/module/kernelmapper.go

```diff
@@ -1,17 +1,41 @@
 package module
 
 import (
+	"encoding/json"
 	"errors"
 	"fmt"
+	"reflect"
 	"regexp"
+	"strings"
 
+	"github.com/a8m/envsubst/parse"
 	ootov1alpha1 "github.com/qbarrand/oot-operator/api/v1alpha1"
+	v1 "k8s.io/api/core/v1"
 )
 
+const (
+	kernelVersionMajorIdx = 0
+	kernelVersionMinorIdx = 1
+	kernelVersionPatchIdx = 2
+)
+
+type NodeOSConfig struct {
+	fullKernelVersion  string `subst:"KERNEL_FULL_VERSION"`
+	kernelVersionMMP   string `subst:"KERNEL_XYZ"`
+	kernelVersionMajor string `subst:"KERNEL_X"`
+	kernelVersionMinor string `subst:"KERNEL_Y"`
+	kernelVersionPatch string `subst:"KERNEL_Z"`
+	kernelRHELVersion  string `subst:"KERNEL_RHEL_VERSION"`
+	kernelRHELRelease  string `subst:"KERNEL_RHEL_RELEASE"`
+	kernelRHELArch     string `subst:"KERNEL_RHEL_ARCH"`
+}
+
 //go:generate mockgen -source=kernelmapper.go -package=module -destination=mock_kernelmapper.go
 
 type KernelMapper interface {
 	FindMappingForKernel(mappings []ootov1alpha1.KernelMapping, kernelVersion string) (*ootov1alpha1.KernelMapping, error)
+	GetNodeOSConfig(node *v1.Node) *NodeOSConfig
```

Comment:
Does it really need to be an interface method? The only one using its return value is `PrepareKernelMapping`.
Wouldn't it be better to make it a private method of the `KernelMapper` object instead?

Reply:
We will probably need the output when we working on MGMT-10539, and probably when I will start looking into CONTAINER_IMAGE template ( currently , not sure if it is needed or not). That's why i made it public, in order for it to be called only once